File: algorithm/src/main/scala/One.scala
import scala.collection._

object One {
    def twoSum(nums : Array[Int], target : Int) : Array[Int] = {
        if (nums.length <= 1) return Array(0, 0)
        val map = mutable.HashMap[Int, Int]()
        for(i <- nums.indices) {
            if(map.contains(nums(i)))  return Array(map(nums(i)), i)
            map.put(target - nums(i), i)
        }
        Array(0, 0)
    }
}

File: algorithm/src/main/java/com/yl/learn/algorithm/hot/One.java
package com.yl.learn.algorithm.hot;

import java.util.HashMap;
import java.util.Map;

public class One {
    
    public int[] twoSum(int[] nums, int target) {
        if(nums == null || nums.length == 0) return null;
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int oth = target - nums[i];
            Integer index = map.get(oth);
            if(index != null) return new int[] {index, i};
            map.put(nums[i], i);
        }
        return null;
    }
}
